compiler/src/openxla/compiler/nvgpu/Conversion/ConvertHLOToCUDNN.cpp

```diff
@@ -124,22 +197,57 @@ static LogicalResult convertClamp(stablehlo::ClampOp op,
     return rewriter.notifyMatchFailure(op, "expected child of graph");
   auto min_or = matchRelu(op, rewriter);
   if (failed(min_or)) return failure();
-  auto operand = op.getOperand();
-  TensorType tensor_type = getCudnnTensorType(operand.getType());
-  auto cast_op = rewriter.create<UnrealizedConversionCastOp>(
-      operand.getLoc(), tensor_type, operand);
+  Type result_type = getCudnnTensorType(op.getType());
+  Value operand = castToCudnnTensor(op.getOperand(), rewriter);
```

Comment:
With dialect conversion (and custom materializations?) I think it will be a bit simpler, no more need for explicit casting for all args

Reply:
I was holding back because dialect conversion itself has some initial cost (you can't mix conversion and ordinary patterns). But yes, it will likely make things easier down the road. I will switch to conversion as a follow-up.